src/SourceCode.Clay.OpenApi/Builders/ComponentsBuilder.cs

```diff
@@ -0,0 +1,122 @@
+{
+    /// <summary>
+    /// Holds a set of reusable objects for different aspects of the OAS. All objects
+    /// defined within the components object will have no effect on the API unless they
+    /// are explicitly referenced from properties outside the components object.
+    /// </summary>
+    public class ComponentsBuilder
+    {
+        #region Properties
+
+        /// <summary>
+        /// Gets the object that holds reusable <see cref="Schema"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<Schema>> Schemas { get; }
+
+        /// <summary>
+        /// Gets the object that holds reusable <see cref="Response"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<Response>> Responses { get; }
+
+        /// <summary>
+        /// Gets the object that holds reusable <see cref="Parameter"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<Parameter>> Parameters { get; }
+
+        /// <summary>
+        /// Gets the object that holds reusable <see cref="Example"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<Example>> Examples { get; }
+
+        /// <summary>
+        /// Gets the object that holds reusable <see cref="RequestBody"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<RequestBody>> RequestBodies { get; }
+
+        /// <summary>
+        /// Gets the object that holds reusable header <see cref="ParameterBody"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<ParameterBody>> Headers { get; }
+
+        /// <summary>
+        /// Gets the object that holds reusable <see cref="SecurityScheme"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<SecurityScheme>> SecuritySchemes { get; }
+
+        /// <summary>
+        /// Gets the object that holds reusable <see cref="Link"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<Link>> Links { get; }
+
+        /// <summary>
+        /// Gets the object that holds reusable <see cref="Callback"/> instances.
+        /// </summary>
+        public IDictionary<string, Referable<Callback>> Callbacks { get; }
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="ComponentsBuilder"/> class.
+        /// </summary>
+        public ComponentsBuilder()
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="ComponentsBuilder"/> class.
+        /// </summary>
+        /// <param name="value">The <see cref="Components"/> to copy values from.</param>
+        public ComponentsBuilder(Components value)
+        {
+            if (ReferenceEquals(value, null)) throw new ArgumentNullException(nameof(value));
+            Schemas = new Dictionary<string, Referable<Schema>>(value.Schemas);
```

Comment:
Please log a bug to use StringComparer.Ordinal where it makes sense (or per OpenApi spec)